### PR TITLE
PDS-3394 bump up hadoop version to 2.10.2

### PR DESCRIPTION
**Pull request by sairajm**
https://jira.infusionsoft.com/browse/PDS-3394

A vulnerability has been identified in version < 2.10.2 of package org.apache.hadoop:hadoop-common in infusionsoft/DataflowTemplates. For more details on the vulnerability see [CVE-2021-37404](https://nvd.nist.gov/vuln/detail/CVE-2021-37404) or the [GitHub Alerts page](https://github.com/infusionsoft/DataflowTemplates/network/alerts).

### CODE REVIEW

**Comment by sairajm** [comment]
Running `mvn -B clean install -f unified-templates.xml -pl v2/cdc-parent/cdc-embedded-connector,v2/streaming-data-generator -am -amd -Dcheckstyle.skip -Dmdep.analyze.skip -Djib.skip` on the tests passes it, the full suite has never been run on this repo and will likely not pass as we are severely out of date. However, I did test that the dataflow job is getting build successfully and is pushed up.

**Comment by sairajm** [comment]
```2022-08-04 11:50:57 INFO  DataflowPipelineTranslator:547 - Adding WriteSuccessfulRecords/StreamingInserts/StreamingWriteTables/GlobalWindow/Window.Assign as step s13
2022-08-04 11:50:57 INFO  DataflowPipelineTranslator:547 - Adding WriteSuccessfulRecords/StreamingInserts/StreamingWriteTables/StripShardId/Map as step s14
2022-08-04 11:50:57 INFO  DataflowPipelineTranslator:547 - Adding WriteSuccessfulRecords/StreamingInserts/StreamingWriteTables/StreamingWrite/BatchedStreamingWrite.ViaBundleFinalization/ParMultiDo(BatchAndInsertElements) as step s15
2022-08-04 11:50:57 INFO  DataflowRunner:1177 - Dataflow SDK version: 2.38.0
2022-08-04 11:50:57 INFO  DataflowRunner:1289 - Printed job specification to gs://is-events-dataflow-intg/templates/pubsub2BGQ.json
2022-08-04 11:50:58 INFO  DataflowRunner:1299 - Template successfully created.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  29.211 s
[INFO] Finished at: 2022-08-04T11:50:58-07:00
```

**Comment by sairajm** [comment]
```[INFO] Skipping plugin execution
[INFO] 
[INFO] --- spotless-maven-plugin:2.12.1:check (default) @ streaming-data-generator ---
[INFO] 
[INFO] --- maven-install-plugin:2.4:install (default-install) @ streaming-data-generator ---
[INFO] Installing /Users/sairaj.madhavan/Documents/projects/DataflowTemplates/v2/streaming-data-generator/target/streaming-data-generator-1.0-SNAPSHOT.jar to /Users/sairaj.madhavan/.m2/repository/com/google/cloud/teleport/v2/streaming-data-generator/1.0-SNAPSHOT/streaming-data-generator-1.0-SNAPSHOT.jar
[INFO] Installing /Users/sairaj.madhavan/Documents/projects/DataflowTemplates/v2/streaming-data-generator/target/dependency-reduced-pom.xml to /Users/sairaj.madhavan/.m2/repository/com/google/cloud/teleport/v2/streaming-data-generator/1.0-SNAPSHOT/streaming-data-generator-1.0-SNAPSHOT.pom
[INFO] Installing /Users/sairaj.madhavan/Documents/projects/DataflowTemplates/v2/streaming-data-generator/target/streaming-data-generator-1.0-SNAPSHOT-sources.jar to /Users/sairaj.madhavan/.m2/repository/com/google/cloud/teleport/v2/streaming-data-generator/1.0-SNAPSHOT/streaming-data-generator-1.0-SNAPSHOT-sources.jar
[INFO] Installing /Users/sairaj.madhavan/Documents/projects/DataflowTemplates/v2/streaming-data-generator/target/streaming-data-generator-1.0-SNAPSHOT-tests.jar to /Users/sairaj.madhavan/.m2/repository/com/google/cloud/teleport/v2/streaming-data-generator/1.0-SNAPSHOT/streaming-data-generator-1.0-SNAPSHOT-tests.jar
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] integration-testing-lib 1.0-SNAPSHOT ............... SUCCESS [ 13.108 s]
[INFO] dynamic-templates 1.0-SNAPSHOT ..................... SUCCESS [ 16.388 s]
[INFO] common 1.0-SNAPSHOT ................................ SUCCESS [ 42.264 s]
[INFO] cdc-parent 0.1 ..................................... SUCCESS [ 13.507 s]
[INFO] cdc-common 0.1 ..................................... SUCCESS [ 20.731 s]
[INFO] cdc-embedded-connector 0.1 ......................... SUCCESS [ 23.694 s]
[INFO] cdc-agg 0.1 ........................................ SUCCESS [ 24.459 s]
[INFO] streaming-data-generator 1.0-SNAPSHOT .............. SUCCESS [ 36.332 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  03:11 min
[INFO] Finished at: 2022-08-04T11:54:53-07:00
[INFO] ------------------------------------------------------------------------
```

**Review by Steven-M-Brown — COMMENTED** [review]
What did you actually change? Getting lost in the codestyle fixes noise.

File: pom.xml [hunk 1]
```diff
@@ -82,7 +82,7 @@
     <tensorflow.version>1.15.0</tensorflow.version>
     <truth.version>1.0.1</truth.version>
     <dlp.version>2.1.0</dlp.version>
-    <hadoop.version>2.8.5</hadoop.version>
+    <hadoop.version>2.10.2</hadoop.version>
```
Comment (sairajm) [inline pom.xml:85]: @Steven-M-Brown  this one

File: pom.xml [hunk 2]
```diff
@@ -98,7 +98,7 @@
       <dependency>
         <groupId>com.google.cloud</groupId>
         <artifactId>libraries-bom</artifactId>
-        <version>24.1.0</version>
+        <version>${cloud-libraries-bom.version}</version>
```
Comment (sairajm) [inline pom.xml:101]: @Steven-M-Brown this one

Comment (sairajm) [reply pom.xml:101]: The previous had a mismatch with the rest of the libraries. As a result, that caused a ton of warnings.

File: v2/pom.xml [hunk 3]
```diff
@@ -31,7 +31,7 @@
         <checkstyle.version>8.7</checkstyle.version>
         <beam.version>2.38.0</beam.version>
         <hamcrest.version>2.1</hamcrest.version>
-        <hadoop.version>2.10.1</hadoop.version>
+        <hadoop.version>2.10.2</hadoop.version>
```
Comment (sairajm) [inline v2/pom.xml:34]: @Steven-M-Brown this one

File: v2/googlecloud-to-googlecloud/pom.xml [hunk 4]
```diff
@@ -35,7 +35,7 @@
     <mockito.version>1.10.19</mockito.version>
     <mockito-core.version>3.0.0</mockito-core.version>
     <derby.version>10.14.2.0</derby.version>
-    <hadoop.version>2.8.5</hadoop.version>
+    <hadoop.version>2.10.2</hadoop.version>
```
Comment (sairajm) [inline v2/googlecloud-to-googlecloud/pom.xml:38]: @Steven-M-Brown this one

File: v2/cdc-parent/cdc-embedded-connector/pom.xml [hunk 5]
```diff
@@ -135,7 +135,7 @@
     <dependency>
       <groupId>org.apache.commons</groupId>
       <artifactId>commons-configuration2</artifactId>
-      <version>2.5</version>
+      <version>2.8.0</version>
```
Comment (sairajm) [inline v2/cdc-parent/cdc-embedded-connector/pom.xml:138]: @Steven-M-Brown this one

**Review by Steven-M-Brown — APPROVED** [review]
None of the changes you called out look too bad, tests are failing though (not sure if they ever passed looking at some history)

**Pull request merged**